sycl/test-e2e/GroupInterface/sub_group.cpp

```diff
@@ -0,0 +1,111 @@
+// REQUIRES: cpu
+
+// RUN: %{build} %cxx_std_optionc++23 -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/detail/core.hpp>
+#define __DPCPP_ENABLE_UNFINISHED_KHR_EXTENSIONS
+#include <sycl/khr/group_interface.hpp>
+
+#include <sycl/builtins.hpp>
+
+#include <type_traits>
+
+using namespace sycl;
+
+static_assert(std::is_same_v<khr::sub_group::id_type, id<1>>);
+static_assert(std::is_same_v<khr::sub_group::linear_id_type, uint32_t>);
+static_assert(std::is_same_v<khr::sub_group::range_type, range<1>>);
+#if defined(__cpp_lib_mdspan)
+static_assert(
+    std::is_same_v<khr::sub_group::extents_type, std::dextents<uint32_t, 1>>);
+#endif
+static_assert(std::is_same_v<khr::sub_group::size_type, uint32_t>);
+static_assert(khr::sub_group::dimensions == 1);
+static_assert(khr::sub_group::fence_scope == memory_scope::sub_group);
+
+int main() {
+  queue q(cpu_selector_v);
+
+  const int sz = 16;
+  q.submit([&](handler &h) {
+    h.parallel_for(nd_range<1>{sz, sz}, [=](nd_item<1> item) {
+      sub_group g = item.get_sub_group();
+
+      khr::sub_group sg = g;
+      assert(sg.id() == g.get_group_id());
```

Comment:
I'm not very confident in using device asserts for our tests... Does anybody know for sure it's stable enough?